### PR TITLE
BookList Added

### PR DESCRIPTION
**Pull request by Nikita5796**

### CODE REVIEW

File: src/app/book/booklist/booklist.component.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  books : Book[] = [];
+
+  ngOnInit() {
+    let categoryid = 1;
+    this.bookService.getAllBooks(categoryid).subscribe((data:Book[]) => {
+      this.books = data;
+    });
+    // console.log(response);
```
Comment (bala0105) [inline src/app/book/booklist/booklist.component.ts]: plz remove commented code

File: src/app/home/home.component.html [hunk 2]
```diff
@@ -5,9 +5,10 @@
     </div>
 
     <div class="card-body">
-      <div class="text-center">
+      <app-booklist></app-booklist>
+      <!-- <div class="text-center">
```
Comment (bala0105) [inline src/app/home/home.component.html]: same here

File: src/app/book/booklist/booklist.component.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  constructor(private bookService: BookService,private config: NgbRatingConfig, private router : Router) { 
+    config.readonly = true;
+    config.max=5;
+  }
+
+  books : Book[] = [];
+
+  ngOnInit() {
+    let categoryid = 1;
```
Comment (bala0105) [inline src/app/book/booklist/booklist.component.ts]: any reason for hard coded value ?

File: src/app/book/booklist/booklist.component.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+          </thead>
+          <tbody>
+            <tr *ngFor = "let book of books" >
+              <td>
+                <a routerLink="">
+                  <img src="{{ book.imageUrl }}" style="width: 70px; height: 90px;" />
+                </a>
+              </td>
+              <td>{{ book.title }}</td>
+              <td>{{ book.price | currency: "USD" }}</td>
```
Comment (ketannaik) [inline src/app/book/booklist/booklist.component.html]: Make it INR instead of USD

File: src/app/book/book.service.ts [hunk 5]
```diff
@@ -20,8 +20,9 @@ export class BookService {
 
   getAllBooks(category : number) {
     console.log(category +" hello");
-    let result =  this.http.get("https://localhost:44356/api/books/category/"+category);
-    console.log(result);
-    return result;
+    let response =  this.http.get("https://localhost:44356/api/books/category/"+category);
+    console.log(response);
```
Comment (ketannaik) [inline src/app/book/book.service.ts]: Remove Console.log

**Review by ketannaik — CHANGES_REQUESTED** [review]

File: src/app/book/book.service.ts (hunk 5 above)
Comment (Nikita5796) [reply src/app/book/book.service.ts]: done.

File: src/app/book/booklist/booklist.component.html (hunk 4 above)
Comment (Nikita5796) [reply src/app/book/booklist/booklist.component.html]: done

File: src/app/book/booklist/booklist.component.ts (hunk 3 above)
Comment (Nikita5796) [reply src/app/book/booklist/booklist.component.ts]: passing category id as 1 to get books of all categories. will work on it and will try to do it in better way.

File: src/app/book/booklist/booklist.component.ts (hunk 1 above)
Comment (Nikita5796) [reply src/app/book/booklist/booklist.component.ts]: done.

File: src/app/home/home.component.html (hunk 2 above)
Comment (Nikita5796) [reply src/app/home/home.component.html]: done.

File: src/app/book/booklist/booklist.component.ts (hunk 3 above)
Comment (ketannaik) [reply src/app/book/booklist/booklist.component.ts]: Technically, you should pass 0 initially which means get books with all categories.

Comment (Nikita5796) [reply src/app/book/booklist/booklist.component.ts]: done.

**Pull request merged**